misk/src/main/kotlin/misk/web/Request.kt

```diff
@@ -3,13 +3,53 @@ package misk.web
 import misk.web.actions.WebSocket
 import okhttp3.Headers
 import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.BufferedSink
 import okio.BufferedSource
 import org.eclipse.jetty.http.HttpMethod
+import java.io.IOException
+import java.net.HttpURLConnection
 
 data class Request(
   val url: HttpUrl,
-  val method: HttpMethod,
+  val method: HttpMethod = HttpMethod.GET,
   val headers: Headers = Headers.of(),
   val body: BufferedSource,
   val websocket: WebSocket? = null
 )
+
+fun Request.toOkHttp3(newUrl: HttpUrl?): Response {
+  val client = OkHttpClient()
+
+  val okUrl = if (newUrl != null) newUrl else this.url
+  val okRequestBody = if (this.method == HttpMethod.GET) null else object : okhttp3.RequestBody() {
```

Comment:
expand this if ? This statement is too awkward for a single line

Reply:
add a TODO to fix the requests to not have a request body if they shouldn’t have a request body? GET is a good approximation here, but in practice it’s much more complicated than that. (HEAD has none, DELETE sometimes has a body, etc.)

Reply:
Good catch, added TODO and issue #279